NullPointerException: Attempt to invoke virtual method 'java.lang.Boolean fr.gaulupeau.apps.Poche.entity.Article.getFavorite()' on a null object

report by @gerroon in https://github.com/wallabag/android-app/issues/203#issuecomment-214517343

my crash


```
I/ActivityManager( 5364): Force stopping fr.gaulupeau.apps.InThePoche appid=10198 user=0: from pid 5364
I/ActivityManager( 5364): Killing 32461:fr.gaulupeau.apps.InThePoche/u0a198 (adj 0): stop fr.gaulupeau.apps.InThePoche
I/WindowState( 5364): WIN DEATH: Window{11ad1dc u0 fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.ArticlesListActivity}
W/ActivityManager( 5364): Force removing ActivityRecord{3895e75f u0 fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.ArticlesListActivity t3454}: app died, no saved state
W/ActivityManager( 5364): Spurious death for ProcessRecord{22cc5e2 0:fr.gaulupeau.apps.InThePoche/u0a198}, curProc for 32461: null
W/PackageManager( 7727): Failure retrieving resources for fr.gaulupeau.apps.InThePoche: Resource ID #0x0
W/PackageManager( 5486): Failure retrieving resources for fr.gaulupeau.apps.InThePoche: Resource ID #0x0
I/log     ( 7727): Launching app://fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.ArticlesListActivity
I/Timeline( 7727): Timeline: Activity_launch_request id:fr.gaulupeau.apps.InThePoche time:63970764
I/ActivityManager( 5364): START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.ArticlesListActivity} from uid 10066 on display 0
V/WindowManager( 5364): addAppToken: AppWindowToken{37566b92 token=Token{1ed11d1d ActivityRecord{1aa6c1f4 u0 fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.ArticlesListActivity t3455}}} to stack=1 task=3455 at 0
V/WindowManager( 5364): Adding window Window{377d2d5 u0 Starting fr.gaulupeau.apps.InThePoche} at 18 of 23 (after Window{36eead01 u0 fr.neamar.kiss/fr.neamar.kiss.MainActivity})
I/ActivityManager( 5364): Start proc 32679:fr.gaulupeau.apps.InThePoche/u0a198 for activity fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.ArticlesListActivity
V/WindowManager( 5364): Adding window Window{ccdd42 u0 fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.ArticlesListActivity} at 18 of 23 (before Window{377d2d5 u0 Starting fr.gaulupeau.apps.InThePoche})
V/WindowManager( 5364): Adding window Window{2cc08a90 u0 fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.ArticlesListActivity} at 18 of 24 (before Window{ccdd42 u0 fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.ArticlesListActivity})
I/ActivityManager( 5364): Displayed fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.ArticlesListActivity: +732ms
I/Timeline( 5364): Timeline: Activity_windows_visible id: ActivityRecord{1aa6c1f4 u0 fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.ArticlesListActivity t3455} time:63971810
V/WindowManager( 5364): Adding window Window{3de026c1 u0 PopupWindow:3d1fa629} at 19 of 23 (after Window{2cc08a90 u0 fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.ArticlesListActivity})
I/Timeline(32679): Timeline: Activity_launch_request id:fr.gaulupeau.apps.InThePoche time:63974205
I/ActivityManager( 5364): START u0 {cmp=fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.SettingsActivity} from uid 10198 on display 0
V/WindowManager( 5364): addAppToken: AppWindowToken{215b1af2 token=Token{3c076dfd ActivityRecord{2f324a54 u0 fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.SettingsActivity t3455}}} to stack=1 task=3455 at 1
V/WindowManager( 5364): Adding window Window{c61ecf9 u0 fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.SettingsActivity} at 20 of 24 (after Window{3de026c1 u0 PopupWindow:3d1fa629 EXITING})
I/ActivityManager( 5364): Displayed fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.SettingsActivity: +384ms
I/Timeline( 5364): Timeline: Activity_windows_visible id: ActivityRecord{2f324a54 u0 fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.SettingsActivity t3455} time:63975018
V/WindowManager( 5364): Adding window Window{ef8bed8 u0 PopupWindow:1d3a0ecd} at 20 of 24 (after Window{c61ecf9 u0 fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.SettingsActivity})
V/WindowManager( 5364): not Base app: Adding window Window{4a6f084 u0 fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.SettingsActivity} at 21 of 25
V/WindowManager( 5364): not Base app: Adding window Window{11e490f0 u0 fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.ArticlesListActivity} at 19 of 24
I/Timeline( 5364): Timeline: Activity_windows_visible id: ActivityRecord{1aa6c1f4 u0 fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.ArticlesListActivity t3455} time:63980363
E/AndroidRuntime(32679): Process: fr.gaulupeau.apps.InThePoche, PID: 32679
E/AndroidRuntime(32679): Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Boolean fr.gaulupeau.apps.Poche.entity.Article.getFavorite()' on a null object referenceE/AndroidRuntime(32679):        at fr.gaulupeau.apps.Poche.network.tasks.UpdateFeedTask.processFeed(UpdateFeedTask.java:347)
E/AndroidRuntime(32679):        at fr.gaulupeau.apps.Poche.network.tasks.UpdateFeedTask.updateByFeed(UpdateFeedTask.java:190)
E/AndroidRuntime(32679):        at fr.gaulupeau.apps.Poche.network.tasks.UpdateFeedTask.updateAllFeeds(UpdateFeedTask.java:124)
E/AndroidRuntime(32679):        at fr.gaulupeau.apps.Poche.network.tasks.UpdateFeedTask.doInBackground(UpdateFeedTask.java:71)
E/AndroidRuntime(32679):        at fr.gaulupeau.apps.Poche.network.tasks.UpdateFeedTask.doInBackground(UpdateFeedTask.java:30)
W/ActivityManager( 5364):   Force finishing activity 1 fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.ArticlesListActivity
W/PackageManager( 7727): Failure retrieving resources for fr.gaulupeau.apps.InThePoche: Resource ID #0x0
W/PackageManager( 5486): Failure retrieving resources for fr.gaulupeau.apps.InThePoche: Resource ID #0x0
I/ActivityManager( 5364): Process fr.gaulupeau.apps.InThePoche (pid 32679) has died
W/PackageManager( 7727): Failure retrieving resources for fr.gaulupeau.apps.InThePoche: Resource ID #0x0
I/ActivityManager( 5364): START u0 {act=android.settings.APPLICATION_DETAILS_SETTINGS dat=package:fr.gaulupeau.apps.InThePoche cmp=com.android.settings/.applications.InstalledAppDetails} from uid 10066 on display 0
W/PackageManager( 7727): Failure retrieving resources for fr.gaulupeau.apps.InThePoche: Resource ID #0x0
^@^@^@^@^@^@^@^@^@y
```

--- a/app/src/main/java/fr/gaulupeau/apps/Poche/network/tasks/UpdateFeedTask.java
+++ b/app/src/main/java/fr/gaulupeau/apps/Poche/network/tasks/UpdateFeedTask.java
@@ -343,6 +343,7 @@ public class UpdateFeedTask extends AsyncTask<Void, Void, Void> {
                     Article article = articleDao.queryBuilder()
                             .where(ArticleDao.Properties.ArticleId.eq(id))
                             .build().unique();
+                    if(article == null) continue;
 
                     if(article.getFavorite() != null && article.getFavorite()) continue;
 

